### PR TITLE
provider mysqlxorm support session for transparent transaction and init-ping

### PR DESCRIPTION
**Pull request by sfwn**
#### What type of this PR

/kind feature

#### What this PR does / why we need it:

- support ping when init to check database advanced
- migrated feature: support session for transparent transaction delivery, see example for usage, which is already used in pipeline

### CODE REVIEW

File: providers/mysqlxorm/examples/main.go [hunk 1]
```diff
@@ -38,6 +39,9 @@ func (p *provider) Init(ctx servicehub.Context) error {
 }
 
 func (p *provider) Run(ctx context.Context) error {
+	/*
+		test db
+	*/
 	r, err := p.MySQL.DB().QueryString("show tables")
```
Comment (Effet) [inline providers/mysqlxorm/examples/main.go:45]: why not "select 1"?

**Review by Effet — CHANGES_REQUESTED** [review]

**Comment by Effet** [comment]
explain why not use gorm.

File: providers/mysqlxorm/examples/main.go (hunk 1 above)
Comment (sfwn) [reply providers/mysqlxorm/examples/main.go:45]: Just an example SQL.

**Comment by sfwn** [comment]
> explain why not use gorm.

We will migrate later. Now xorm provider is used by pipeline&qa&hepa. So this provider is required.

**Review by Effet — APPROVED** [review]

**Pull request merged**